### PR TITLE
Create palindrome.dart

### PR DESCRIPTION
**Pull request by satyamnoob**
# Welcome to Dart community

### **Describe your change:**

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Dart files are placed inside an existing directory.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by satyamnoob** [comment]
please add the label 'hacktoberfest-accepted' when accepting the pr request

**Review by akashgk — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: strings/palindrome.dart [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+print('Enter Words or number');
+    // User enter a string or a number
+    String? original =  'hannah';
+  
+   // then we will reverse the input
+   String? reverse = original.split('').reversed.join('');
+  // then we will compare
+  if(original == reverse)
+  {
+    print('Its A Palindrome');
```
[the remark this answers is not in the document]
Comment (satyamnoob) [reply strings/palindrome.dart:11]: I am very new to this. What are tests? I will be updating everything as you requested

[the remark this answers is not in the document]
Comment (akashgk) [reply strings/palindrome.dart:11]: You can check out other files. to see how tests are implemented. 


You can also read up from the official docs:
https://pub.dev/packages/test

**Comment by satyamnoob** [comment]
I have added the tests and other requirements. Do I need to do another pull request or you can see the changes? Sorry for asking dumb questions

File: strings/palindrome.dart [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  } else {
+    // print('Its A Not Palindrome');
+  }
+
+  if (original == reverseUsingCodeUnits) {
+    // print('Its A Palindrome');
+  } else {
+    // print('Its A Not Palindrome');
+  }
+
```
Comment (akashgk) [inline strings/palindrome.dart:42]: What's the use of this code section?

File: strings/palindrome.dart [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    });
+    test('Using runes to reverse a string.', () {
+      var string = 'hannah';
+      expect(string, equals(reverseStringUsingRunes(string)));
+    });
+    test('Using code units to reverse a string.', () {
+      var string = 'hannah';
+      expect(string, equals(reverseStringUsingCodeUnits(string)));
+    });
+  });
```
Comment (akashgk) [inline strings/palindrome.dart:61]: add negative cases

File: strings/palindrome.dart [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+String reverseStringUsingRunes(String input) {
+  var chars = input.runes.toList();
+
+  return String.fromCharCodes(chars.reversed);
+}
+
+String reverseStringUsingCodeUnits(String input) {
+  return String.fromCharCodes(input.codeUnits.reversed);
+}
```
Comment (akashgk) [inline strings/palindrome.dart:85]: All of the reversal logic are running on O(N) space. can we optimise to make it run on O(1) space.

**Review by akashgk — CHANGES_REQUESTED** [review]
Please address the change requests.

**Pull request closed without merging**